scalyr_agent/log_processing.py

```diff
@@ -750,6 +738,39 @@ def readline(self, current_time=None):
                     result.timestamp = timestamp
                     if attrs:
                         result.attrs = attrs
+                if self.__merge_json_parsed_lines:
+                    while result.line and not result.line.endswith(b"\n"):
```

Comment:
(just thinking out loud) This should also work correctly if lines / messages contain carriage return since that's always presents between line feed character, right?

Reply:
As far as I'm aware `\r` will be before `\n` if it is present at all yes.